Utils/src/GenParticlesProducer.cc

```diff
@@ -302,42 +304,66 @@ void GenParticlesProducer::storeMinimal(const edm::Handle< edm::View<reco::GenPa
                                         std::unique_ptr<std::vector<int>>& ParentId_vec) const {
     std::unordered_set<const reco::Candidate *> stored_particles_ref;
     std::vector<const reco::Candidate *> stored_particles_list, parents_list;
-    for(const auto& iPart : *genPartCands) {
+    for(const auto& particle : *genPartCands) {
         // Skip particles which are already in the list of stored particles
         // This is to protect against parts of a chain which may be saved, but here we are looking at the first copy
-        if(stored_particles_ref.find(&iPart)!=stored_particles_ref.end()) continue;
+        if(stored_particles_ref.find(&particle)!=stored_particles_ref.end()) continue;
+
+        // Store the initial partons
+        auto nmothers = particle.numberOfMothers();
+        if(nmothers == 1) {
+           auto mother = particle.mother(0);
+           if(mother->pdgId() == proton &&
+              std::binary_search(proton_status_codes.begin(), proton_status_codes.end(), mother->status()) &&
```

Comment:
as long as there's only one status code, `binary_search` is probably overkill - just compare to it directly

Reply:
Done